python/shotgun_globals/cached_schema.py

```diff
@@ -881,3 +881,27 @@ def get_ordered_status_list(cls, display_names=False, project_id=None):
         else:
             return self._status_data[project_id]["status_order"]
 
+    @classmethod
+    def clear_cached_data(cls, project_id=None):
+        """
+        Remove both the schema and status cache files from disk for
+        the specified project_id. If no project_id is specified, then
+        use the current context project.
+
+        :param project_id: The id of the project entity to remove
+                           schema and status cache files for. If
+                           None, the current context's project will
+                           be used.
+        """
+        self = cls.__get_instance()
+        project_id = project_id or self._get_current_project_id()
+
+        schema_cache = self._get_schema_cache_path(project_id)
+        if os.path.isfile(schema_cache):
+            self._bundle.log_debug("Removing schema cache file : %s" % schema_cache)
+            os.remove(schema_cache)
```

Comment:
Let's log an error for these if they fail:

```
try:
    os.remove(schema_cache)
except Exception:
    self._bundle.log_error( ... )
    raise
```

Unfortunately, it looks like it will need to be `Exception` that we catch instead of `OSError`, as we get a `WindowsError` on Windows, but that exception doesn't exist for non-Windows OSs. Meh.